File: app/javascript/helpers/show_until_validator.js
export default class ShowUntilValidator {
    static error = "Время актуальности новости должно быть заполнено и быть больше текущего времени";

    static validate(vueState, time) {
        if (+time < +new Date()) {
            vueState.$data.news.errors.push(ShowUntilValidator.error);
            return false;
        } else {
            vueState.$data.news.errors = vueState.$data.news.errors.filter((error) => {
                return error !== ShowUntilValidator.error;
            });

            return true;
        }
    }
}

File: app/validations/show_until_validator.rb
class ShowUntilValidator < ActiveModel::EachValidator
  def validate_each(record, attribute, value)
    if value.blank? || (value.present? && value < Time.zone.now)
      record.errors[attribute] <<
        I18n.t('activerecord.errors.models.news.attributes.show_until.wrong_time')
    end
  end
end